src/main/java/com/booleanuk/Poker.java

```diff
@@ -1,18 +1,122 @@
+
+        pair = false;
+        threeOfAKind = false;
+        fourOfAKind = false;
+        fullHouse = false;
+    }
+
+    public boolean isPair(String[] hand){
+        int i = 0;
+        return hand[i].equals(hand[i + 1]);
+    }
+
     public String[] winningPair(String[] firstHand, String[] secondHand) {
         // Implement the winningPair logic here and return the array containing the winning pair to make the tests pass.
         // You can replace the following return value with something appropriate
-        return new String[]{"Replace me", "with something else"};
+        int totalvalue1 = 0;
+        int totalvalue2 = 0;
+        int index = 0;
+        String[] noPair = new String[0];
+
+        if (firstHand == null && secondHand == null){
+            return noPair;
+        }else if (firstHand == null && secondHand != null){
+            return secondHand;
+        }else if (firstHand != null && secondHand == null){
+            return firstHand;
+        }
+
+        if (isPair(firstHand) && isPair(secondHand)) {
+            for (String string : firstHand) {
+                totalvalue1 += cardValue.get(string);
+            }
+            for (String string : secondHand) {
+                totalvalue2 += cardValue.get(string);
+            }
+            if (totalvalue1 > totalvalue2){
+                return firstHand;
+            }
+            else{
+                return secondHand;
+            }
+        }
+
+        else if (isPair(firstHand) && !isPair(secondHand)) {
+            return firstHand;
+        }
+
+        else if (isPair(secondHand) && !isPair(firstHand)) {
+            return secondHand;
+        }
+
+        return noPair;
+
+        /*
+        else if (!isPair(firstHand) && !isPair(secondHand)){
+            for (String string : firstHand) {
+                totalvalue1 += cardValue.get(string);
+            }
+            for (String string : secondHand) {
+                totalvalue2 += cardValue.get(string);
+            }
+        }
+        if (totalvalue1 > totalvalue2){
+            return firstHand;
+        }
+        else{
+            return secondHand;
+        }
+
+         */
     }
 
     // Extension 1
     public String[] winningPairFromArray(String[][] hands) {
         // Implement the winningPairFromArray logic here and return the array containing the winning pair to make the tests pass.
         // You can replace the following return value with something appropriate
-        return new String[]{"Replace me", "with something else"};
+        String [] winningHand = hands[1];
+        String [] currentHand;
+
+        for (String[] hand : hands) {
+            if (isPair(hand)) {
+                currentHand = hand;
+                winningHand = winningPair(winningHand, currentHand);
+            }
+        }
+        if (isPair(winningHand)){
+            return winningHand;
+        }
+        return new String[]{};
     }
 
     // Extension 2
```

Comment:
Your use of a hashmap to store card values and the implementation of hand comparison logic are effectively executed. However, for full marks you need to have a go at ext 2.